### ISSUE TITLE
Being able to add new types of contributors and have the list alphabetized

### ISSUE THREAD

**Issue by BrittanyBunk** [internetarchive/openlibrary#2383]
There are so many types of contributors not on the list that should be there, but it's really hard to add in. It would be easier to add them from the drop-down menu. Also, it's so hard to find the title, so if it's alphabetized, it'd be much easier.

**Comment by seabelis** [comment]
Related to https://github.com/internetarchive/openlibrary/issues/2241

**Comment by BrittanyBunk** [comment]
this doesn't relate to editions. It's the list that editors choose the contributor's title for. It's a drop down menu. It's out of order and is possible to add more (like assistant designer, etc.), but it's a large hassle. Since people have to manually add them in, the list doesn't get alphabetized.

**Comment by cdrini** [comment]
I think an autocompleting search box with the option to add a new item might be the best UX. I think adding a new option used to be possible, but was disabled at some point (for some reason).

**Comment by BrittanyBunk** [comment]
Great!

**Comment by guyjeangilles** [comment]
I'm cautions to allow users to create roles. What if two users and synonyms in different languages. I.e "Cover Designer" and "Créateur de Couverture". How would this system handle that?

**Comment by tfmorris** [comment]
> I'm cautions to allow users to create roles. What if two users and synonyms in different languages. I.e "Cover Designer" and "Créateur de Couverture". How would this system handle that?

You're right to be cautious. The previous system would just allow duplicates to be created -- and it's not a language thing. You could also end up with "Cover Designer" and "Cover Illustrator." 

If we were to crowd source roles, a more reasonable flow would be 1) suggest "Foo" as a new role, 2) vote on "Foo" as a valid new role, 3) evaluate the vote and either: a) automatically promote (>N votes, > X/Y positive votes), b) automatically reject proposal or c) kick to admins for adjudication.

**Comment by BrittanyBunk** [comment]
@tfmorris I think it's better if members can add new roles and it does show up right away, so that the necessary info's added in then and there. It could be sent after that to a voting area and checked if it's appropriate. I would say a better method is just to be able to flag them to report it and have it removed. 

I think it's good to have specifics, like  "Cover Designer" and "Cover Illustrator", as it should be named according to what the book says. If not, we're just guessing at which place the person fits, and we may get it wrong.
